platform/ios/src/MGLMapView.mm

```diff
@@ -3022,45 +3026,31 @@ - (void)annotationImageNeedsRedisplay:(MGLAnnotationImage *)annotationImage
 
 #pragma mark - User Location -
 
-- (void)setShowsUserLocation:(BOOL)showsUserLocation
+- (void)validateLocationServices
 {
-    if (showsUserLocation == _showsUserLocation || _isTargetingInterfaceBuilder) return;
-
-    _showsUserLocation = showsUserLocation;
+    BOOL shouldEnableLocationServices = !self.locationManager && self.showsUserLocation;
```

Comment:
Validation methods shouldn’t have preconditions. This flag should additionally check for `self.dormant`, such that you could call this method twice in a row without producing the opposite result.